src/SymfonyCache/HeaderReplaySubscriber.php

```diff
@@ -147,4 +162,31 @@ private function checkRequest(Request $request)
 
         return true;
     }
+
+    /**
+     * Checks if at least cookies were provided at all and if at least one
+     * is relevant (not ignored) for a preflight request.
+     *
+     * @param ParameterBag $cookies
+     *
+     * @return bool
+     */
+    private function hasRelevantCookie(ParameterBag $cookies)
+    {
+        $count = $cookies->count();
+
+        if (0 === count($this->options['ignore_cookies'])) {
+            return 0 !== $count;
+        }
+
+        foreach ($cookies as $name => $value) {
+            foreach ($this->options['ignore_cookies'] as $rgxp) {
+                if (!preg_match($rgxp, $name)) {
+                    return true;
```

Comment:
Does this actually work? If I have two `ignore_cookies`, both of them would have to match every cookie with this code I think.

Reply:
True, fixed and added quite some more unit test cases in https://github.com/terminal42/header-replay-bundle/pull/5/commits/b25135ea12e816a3fb53e4dd28e77c07eb51b769.